src/form/src/components/Paged.test.js

```diff
@@ -127,4 +134,37 @@ describe('Paged', () => {
       expect(wrapper.find('input#root_section1_name').props().value).toEqual('name');
     });
   });
+
+  describe('setPage', () => {
+    test('should set schema/uiSchema to the new page', () => {
+      const wrapper = mount(
+        <Paged
+            render={mockRender} />
+      );
+
+      // go to page 3
+      wrapper.find('#setPageButton').simulate('click');
+
+      expect(wrapper.find(Form).props().schema).toEqual(schemas[2]);
+      expect(wrapper.find(Form).props().uiSchema).toEqual(uiSchemas[2]);
+    });
+  });
+
+  test('should persist state', () => {
+    const wrapper = mount(
+      <Paged
+          render={mockRender} />
+    );
+
+    wrapper.find('input#root_section1_name').props().onChange({ currentTarget: { value: 'name' } });
+
+    // to go page 3
+    wrapper.find('#setPageButton').simulate('click');
+
+    // to to page 0
+    wrapper.find('#back-button').simulate('click');
+    wrapper.find('#back-button').simulate('click');
+
+    expect(wrapper.find('input#root_section1_name').props().value).toEqual('name');
+  });
```

Comment:
I think you mean for this test to be inside the 'setPage' suite right?

Reply:
Right! fixing